packages/front-end/components/AutoGenerateMetricsModal.tsx

```diff
@@ -49,14 +57,21 @@ export default function AutoGenerateMetricsModal({
   }>({
     defaultValues: {
       datasourceId: datasource?.id || "",
+      schema: "",
       settings: datasource?.settings || {},
       metricsToCreate: [],
     },
   });
 
+  const selectedSchema = form.watch("schema");
+
   const selectedDatasource =
     datasource || getDatasourceById(form.watch("datasourceId"));
 
+  const { data: selectedDatasourceData } = useApi(
+    `/datasource/${selectedDatasource?.id}/schema`
```

Comment:
This will potentially make an API call to `/datasource//schema` (if `selectedDatasource` is null)

Reply:
@jdorn Good catch re: mentions of "Data Sets". I've updated this so it supports the different schema names (e.g. BQ using Datasets, vs Athena using "Catalogs".) Now, all of the available warehouse types that have auto metric support from an event tracker are handled cleanly. We'll likely want to refactor this logic as the list of supported event trackers expands. I'm thinking we may want to build some sort of map that we can re-use. But that might be overkill in this situation.

I have tested this with Segment + Postgres, and I haven't found any bugs. But, my Postgres instance isn't seeded with events from Segment, it's all just mocked as if it were.